packages/components/src/utils/math.js

```diff
@@ -19,26 +19,34 @@ export function getNumber( value ) {
 /**
  * Safely adds 2 values.
  *
- * @param {number|string} args Values to add together.
+ * @param {Array<number|string>} args Values to add together.
  *
  * @return {number} The sum of values.
  */
 export function add( ...args ) {
-	return args.reduce( ( sum, arg ) => sum + getNumber( arg ), 0 );
+	return args.reduce(
+		/** @type {(sum:number, arg: number|string) => number} */
+		( sum, arg ) => sum + getNumber( arg ),
+		0
+	);
 }
 
 /**
  * Safely subtracts 2 values.
```

Comment:
@ItsJonQ Is this function helpful? Typically you don't see subtract methods on an arbitrary number of elements, probably because it's not an associative operation.

The implementation could be vastly simplified if this were actually designed to work on 2 arguments like the description suggests.

Looking at https://github.com/WordPress/gutenberg/pull/23006, it doesn't seem like there was a need for this to operate on an arbitrary number of elements.

Reply:
@sirreal I think this function is really helpful 🙏 .

Lately, I've been relying on these functions for `Input` related adjustments in G2.
The annoying thing about input values is they can be strings numeric values, or nulls, empty strings, etc...

Having something that can reliably add (assumed) numeric values together is really helpful.
The previous implementation only supported 2 values, so that description is out of date 🙈 .
Thank you for catching that.

P.S. If you know of a better way to reliably handle MATH operations for assumed number types... I'm all ears <3!

Reply:
> I think this function is really helpful 🙏 .

I'm on board with functions providing the utility we need, I wasn't questioning the utility of a subtraction function that can operate on numbers or number-like string values, I'll clarify below 😉 

> If you know of a better way to reliably handle MATH operations for assumed number types...

I'm in favor of keeping things small and dependencies minimal, but this module already depends on lodash and it could pull in a few more things. In the context of WordPress, that will actually reduce the bundle size because lodash will be available in its entirety.

Some of these are also very tricky tricky to get right and lodash is battle tested:

- `getNumber` -> [`_.toNumber`](https://lodash.com/docs/4.17.15#toNumber)
- `add` -> [`( values ) => _.sumBy( values, _.toNumber )`](https://lodash.com/docs/4.17.15#sumBy)

> Having something that can reliably add [any number of] (assumed) numeric values together is really helpful.

Yep, this is perfectly fine. I have no concerns about `add`. It's `subtract` that concerns me. This may seem pedantic, but bear with me 😅  I wonder specifically about subtraction of an arbitrary number of elements which seems strange. A variadic subtraction function (subtract an arbitrary number of values) is a bit odd. It's important that we subtract _from_ a number or [things get weird](https://www.typescriptlang.org/play?#code/PQKhCgAIUgFBDATgZwKbMvAdgE0o1AF0QEtUA3dTSLAVwFsAjVRSc+AG1tQDooJokAAIAHJPHqQA3rSwBrLAHsA7lgC+bTt0gAVABapIJLAGNF9YwHNNXXv37CChWoizS6TFhv2GxKVHgezKzstnzQwOAAZrImhCSKbpZEAHIMwQAUodwAXJDIxFaQAD406SwAlNLgAJBmWAVlnqwAvJBpzVlaqBUA3OC1Ti5uJMgp8CkZQZWQAPyQAAyQedOI-WoDoAIwAMrwUagcAJ75tIzE8HEYAEw23Mjh28J+EtIAgoiI8EcAPKvFBVIWEsAD4NEhLBgAGrdDCERSnc5fOKQeHJQgGRCPBxCIaudzlRDeAyQHAkKIHAimQyKKKoknZdCPSIxUzxRKIi5xDI8XkQ5B5D5fX6AoqlVYgqpSQZEYaYRCQngEHC0EyoDK1GoZMkUlaEgA08sseVFwJKTWChuMOFQAA8KnrmpAWiDqjV3fVGoznZB0R1MhC+pqaniRrg7c6Wm0lvNvXkdXSALR3VD9d1qfWaha1IMbaKxdlueA4HA8vkKgWQIXfH6m6ziwmSt2ho0PZWq9Wa7XkqKOy1Gk2FM0N5pW8P2vssZ2u6XuuqJL3dH1+wkZQNp90thOQADUKY3GazOfWm2AukxqAA5BhkOYiHoisgRKgTCROJhGIpaIROcjCzxIAAdUMLBUACVEEQKJAf3gGDIAAfQsOhUFweDIEUVgMSsDAcESS8f3oeA5EMNAGjsT1FA4XgOEUSwMmQM4uUIDIKgqcBgDPJZAF4NwAhPcgAB5ABpcAKKongaLohikUuZiAEYFlY9iz3kyAuMAVD3IAATVQeA9EAMgIRIXSjqNo+jGL-MzpO5CpDXkxSOMgRMVK4wAaPcgZRUBIRAcAM0STMk8yZMspjgos+TDSkkLWJsyA7LYhzuLcjyvJ809IAASTceoLgKQ1izJQsjAwAArWhGkuEwGFoDhYKKeCGPoQjcGQeDDIaYzxNM-KWPizjIEAUHI2tvMSJLXEsMjipTYqWQa-M6ujuu6mLursvLxpWhToqm64ZvAIA):

```ts
// There's something special about subtraction. We need to start at a _minuend_ or things don't make sense.
console.log(subtract())
// 0 🙂 OK
console.log(subtract(10))
// 10 😕 Yeah…
console.log(subtract(subtract(), 10))
// -10 😬 weird…
console.log(subtract(subtract(subtract(10, subtract())), 10))
// 0 😬 weird…

// In contrast, addition is just accumulating _summands_
console.log(add())
// 0 ✅
console.log(add(10))
// 10 ✅
console.log(add(add(), add(10), add(add(10))))
// 20 ✅
```

You might also consider that the result of subtraction the difference. What is the difference of 3 things?

If you look, you can see this detail buried in the current implementation. Look at line 40 here:

https://github.com/WordPress/gutenberg/blob/2cf608a2095f35f495f189cb2cfcd4f62631c247/packages/components/src/utils/math.js#L37-L42

There's an exception: don't subtract the first value, just pass it through. If we rethink this as a function that takes a minuend (our strating point) and we subtract a variadic number of subtrahends from it (0 or more) it makes more sense. Our function returns the difference of the minuend and the sum of the subtrahends:

```ts
function subtract( minuend: string|number, ...subtrahends: Array<string | number>): number {
	return getNumber( minuend ) - add( ...subtrahends );
}
```

So, this is a little bit pedantic, the result is the same, but it makes more sense to me. There's a difference (no pun intended) between the first value and the rest of the values that's significant.

Reply:
…this is also a bit off topic here. I'll open a PR with some of the suggested changes and the conversation can continue there.